isis/src/odyssey/apps/thm2isis/thm2isis.cpp

```diff
@@ -0,0 +1,268 @@
+#include "LineManager.h"
+#include "UserInterface.h"
+#include "CubeAttribute.h"
+#include "FileName.h"
+#include "IException.h"
+#include "iTime.h"
+#include "ProcessByBrick.h"
+#include "Brick.h"
+#include "OriginalLabel.h"
+
+using namespace std;
+
+namespace Isis { 
+  static vector<Cube *> outputCubes;
+  static int frameletLines;
+  static void TranslateLabels(Pvl &labelFile, Pvl &isis3, int numBands, UserInterface &ui);
+  static void separateFrames(Buffer &in);
+
+
+  void thm2isis(UserInterface &ui) {
+    // Grab the file to import
+    ProcessImportPds p;
+    outputCubes.clear();
+    frameletLines = 192;
+
+    FileName in = ui.GetFileName("FROM");
+
+    // Make sure it is a Themis EDR/RDR
+    bool projected;
+    try {
+      Pvl lab(in.expanded());
+      projected = lab.hasObject("IMAGE_MAP_PROJECTION");
+      QString id;
+      id = (QString)lab["DATA_SET_ID"];
+      id = id.simplified().trimmed();
+      if(!id.startsWith("ODY-M-THM")) {
+        QString msg = "Invalid DATA_SET_ID [" + id + "]";
+        throw IException(IException::Unknown, msg, _FILEINFO_);
+      }
+    }
+    catch(IException &e) {
+      QString msg = "Input file [" + in.expanded() +
+                    "] does not appear to be " +
+                    "in Themis EDR/RDR format";
+      throw IException(IException::Io, msg, _FILEINFO_);
+    }
+
+    //Checks if in file is rdr
+    if(projected) {
+      QString msg = "[" + in.name() + "] appears to be an rdr file.";
+      msg += " Use pds2isis.";
+      throw IException(IException::User, msg, _FILEINFO_);
+    }
+
+    // Ok looks good ... set it as the PDS file
+    Pvl pdsLab;
+    p.SetPdsFile(in.expanded(), "", pdsLab);
+
+    OriginalLabel origLabels(pdsLab);
+
+    Pvl isis3Lab;
+    TranslateLabels(pdsLab, isis3Lab, p.Bands(), ui);
+
+    // Set up the output cube
+    FileName outFile(ui.GetFileName("TO"));
+    PvlGroup &inst = isis3Lab.findGroup("Instrument", Pvl::Traverse);
+    CubeAttributeOutput outAttr = ui.GetOutputAttribute("to");
+    
+    if((QString)inst["InstrumentId"] == "THEMIS_VIS") {
+      Cube *even = new Cube();
+      Cube *odd = new Cube();
+
+      even->setDimensions(p.Samples(), p.Lines(), p.Bands());
+      odd->setDimensions(p.Samples(), p.Lines(), p.Bands());
+      
+      // even->setPixelType(Isis::Real);
+      // odd->setPixelType(Isis::Real);
+
+      QString evenFile = outFile.path() + "/" + outFile.baseName() + ".even.cub";
+      QString oddFile = outFile.path() + "/" + outFile.baseName() + ".odd.cub";
+
+      even->create(evenFile, outAttr);
+      odd->create(oddFile, outAttr);
+
+      frameletLines = 192 / ((int)inst["SpatialSumming"]);
+
+      outputCubes.push_back(odd);
+      outputCubes.push_back(even);
+    }
+    else {
+      Cube *outCube = new Cube();
+      outCube->setDimensions(p.Samples(), p.Lines(), p.Bands());
+      
+      // outCube->setPixelType(Isis::Real);
+
+      outCube->create(outFile.expanded(), outAttr);
```

Comment:
This is basically the bug fix (hard to find when you also do a app convert). Output attributes passed into `Cube::create`.